src/runtime-pmi.c

```diff
@@ -123,6 +123,14 @@ shmem_runtime_fini(void)
         free(location_array);
     }
 
+    if (kvs_key) {
```

Comment:
This one is also leaked:
```
if (kvs_name) free(kvs_name);
```

Reply:
Yes indeed, thanks.

I suppose we don't need the non-null checks on these pointers either, right?  I almost forgot `free` does no action on a NULL pointer.

Reply:
That's a good point. As far as I can tell, this behavior from `free` is portable.